packages/cdk-assets/lib/private/util.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Creates a critical section, ensuring that at most one function can
+ * enter the critical section at a time.
+ */
+export function createCriticalSection() {
+  let lock = Promise.resolve();
+  return async (criticalFunction: () => Promise<void>) => {
+    const res = lock.then(() => criticalFunction());
+    lock = res.catch(e => e);
+    return res.then(v => v);
```

Comment:
Couldn't this be `return res` ? What's the value of the added `then` ?

Reply:
Good spotting this. I've removed it - it was a leftover from some hand testing I was doing.